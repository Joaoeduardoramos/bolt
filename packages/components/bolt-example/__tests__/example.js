import {
  isConnected,
  render,
  renderString,
  stopServer,
  html,
} from '../../../testing/testing-helpers';

describe('example', () => {
  afterAll(async () => {
    await stopServer();
  }, 100);

  test(`example with component`, async () => {
    const results = await render('@bolt-components-example/example.twig', {
      title: 'Example Title',
      content:
        '<p>Nulla facilisi. Praesent ac massa at ligula laoreet iaculis. Vivamus quis mi. Praesent porttitor, nulla vitae posuere iaculis, arcu nisl dignissim dolor, a pretium mi sem ut ipsum. Sed aliquam ultrices mauris.</p>',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
