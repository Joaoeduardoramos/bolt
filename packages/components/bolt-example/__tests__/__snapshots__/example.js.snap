// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example example with component 1`] = `
<bolt-example>
  <replace-with-grandchildren class="c-bolt-example c-bolt-example--spacing-medium">
    <div class="c-bolt-example__title">
      <div slot="title">
        Example Title
      </div>
    </div>
    <div class="c-bolt-example__content">
      <p>
        Nulla facilisi. Praesent ac massa at ligula laoreet iaculis. Vivamus quis mi. Praesent porttitor, nulla vitae posuere iaculis, arcu nisl dignissim dolor, a pretium mi sem ut ipsum. Sed aliquam ultrices mauris.
      </p>
    </div>
  </replace-with-grandchildren>
</bolt-example>
`;
